apps/gitness/src/pages-v2/repo/repo-commits.tsx

```diff
@@ -24,7 +25,22 @@ export default function RepoCommitsPage() {
     queryParams: { page }
   })
 
-  const [selectedBranch, setSelectedBranch] = useState<BranchSelectorListItem>({ name: '', sha: '' })
+  const { branchList, setBranchList, selectedBranch, setSelectedBranch, setRepoId, setSpaceId } =
+    useBranchSelectorStore()
+
+  useEffect(() => {
+    branches &&
+      !branchList &&
+      setBranchList(
+        branches.map(item => ({
+          name: item.name || '',
+          sha: item.sha || ''
+        }))
```

Comment:
should this not have `divergence` included?

Reply:
Branch Selector isn't using it. We may add more props as and when needed, right?